enumerable.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < check_self.length
+      yield check_self[i]
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum :my_each_with_index unless block_given?
+
+    i = 0
+    my_each do |element|
+      yield element, i
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return enum_for :my_select unless block_given?
+
+    Enumerator.new do |y|
+      my_each { |item| y << item if yield item }
+    end.to_a
+  end
+
+  def my_all?(initial = nil)
+    result = true
+    if block_given?
+      my_each { |element| result &= (yield element) }
+    elsif initial
+      my_each { |element| result &= pattern == element }
+    else
+      my_each { |element| result &= element }
+    end
+    result
+  end
+
+  def my_any?
+    my_each { |item| return true if block_given? ? yield(item) : item }
+    false
+  end

```

Comment:
test my_any? against:

 ```
%w[ant bear cat].any? { |word| word.length >= 3 } #=> true
%w[ant bear cat].any? { |word| word.length >= 4 } #=> true
%w[ant bear cat].any?(/d/)                        #=> false
[nil, true, 99].any?(Integer)                     #=> true
[nil, true, 99].any?                              #=> true
[].any?                                           #=> false
```

- [ ] my_any? returns true if the block ever returns a value other than false or nil
- [ ] my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
- [ ] my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class
- [ ] my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex